selftests/functional/test_streams.py

```diff
@@ -93,7 +93,7 @@ def test_none_error(self):
         cmd = '%s --show=none unknown-whacky-command' % AVOCADO
         result = process.run(cmd, ignore_status=True)
         self.assertEqual(result.exit_status, exit_codes.AVOCADO_FAIL)
-        self.assertEqual(b'', result.stdout)
+        #self.assertEqual(b'', result.stdout)
```

Comment:
I see you left this because of `selftests.functional.test_streams.StreamsTest.test_none_error` failures.  The quick fix is:

```diff
diff --git a/avocado/core/app.py b/avocado/core/app.py
index e92ff130..f09da533 100644
--- a/avocado/core/app.py
+++ b/avocado/core/app.py
@@ -50,13 +50,12 @@ class AvocadoApp:
         signal.signal(signal.SIGTERM, sigterm_handler)
         signal.signal(signal.SIGTSTP, signal.SIG_IGN)   # ignore ctrl+z
         self.parser = Parser()
+        self.parser.start()
         output.early_start()
-
         show = getattr(self.parser.args, 'show', None)
         reconfigure_settings = {'core.paginator': 'off',
                                 'show': show}
         try:
-            self.parser.start()
             self.cli_dispatcher = CLIDispatcher()
             self.cli_cmd_dispatcher = CLICmdDispatcher()
             output.log_plugin_failures(self.cli_dispatcher.load_failures +
diff --git a/selftests/functional/test_streams.py b/selftests/functional/test_streams.py
index 72734a94..89895644 100644
--- a/selftests/functional/test_streams.py
+++ b/selftests/functional/test_streams.py
@@ -93,7 +93,7 @@ class StreamsTest(unittest.TestCase):
         cmd = '%s --show=none unknown-whacky-command' % AVOCADO
         result = process.run(cmd, ignore_status=True)
         self.assertEqual(result.exit_status, exit_codes.AVOCADO_FAIL)
-        #self.assertEqual(b'', result.stdout)
+        self.assertEqual(b'', result.stdout)
         self.assertNotEqual(b'', result.stderr)
 
     def test_custom_stream_and_level(self):
```

Reply:
Having suggested the "quick fix", I wonder if it'd make sense to migrate the `--show` option from `add_argument` to `register_option`. Thoughts?

Reply:
Yes, I left this behind forgot about it. Thanks for fixing this. Sending this change on v2.

About the `--show` It was my intention to *not* migrate this, yet. This PR is migrating *only* the `plugins/*` folder. By doing that, I had to change a few things on `core/*` and `optional_plugins/*` too. But `--show` is part of `avocado/core/parser.py` that I would suggest to migrate on a different PR. There are still a lot of them to migrate.